### ISSUE TITLE
Dashboard: Improve UX for stories without edit link

### ISSUE THREAD

**Issue by spacedmonkey** [GoogleForCreators/web-stories-wp#8652]
## Feature Description
 
In WordPress, there are different user levels. The different user levels do not access to edit all stories. In the dashboard, a list of stories are shown. In https://github.com/google/web-stories-wp/pull/8591, the options to edit the story were hidden. However, the title of the story, styled to look like a hyperlink. 

![Screenshot 2021-08-11 at 12 18 46](https://user-images.githubusercontent.com/237508/129020149-8e33589b-7aae-4c29-bc9f-7216082a6e89.png)
In this above, screenshot, the user do not have access to edit "test admin". There is no way to see that. 

Compare to WordPress's UX. 

![Screenshot 2021-08-11 at 12 19 48](https://user-images.githubusercontent.com/237508/129020271-d1c52440-1aec-47d0-b163-2c51cc5f044d.png)

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

**Comment by swissspidy** [comment]
As mentioned on the PR, an alternative would be to just remove that link in any case. A story can be edited via the three dots menu. On the other hand, it's a nice quick way to edit a story.

cc @o-fernandez @andrew-askedall for thoughts on UX here.

**Comment by o-fernandez** [comment]
My thought was to... 
- Check if the user can edit a story
- If not, make the title not-clickable and disable the "open in editor" option from the 3-dot menu

Would that work?

**Comment by BrittanyIRL** [comment]
> * If not, make the title not-clickable and disable the "open in editor" option from the 3-dot menu

I suppose a little more context here is helpful for you :) 

So, Jonny just updated the menu to hide the context menu actions that a user cannot do (say, open the story in editor). 
The issue here is that historically the title of the story that is visible on top of the story poster is a link to open the story in the editor. The problem is making sure that that title is _visibly_ a link when it's clickable and _visibly_ not a link when it's not. 

The whole reason the title was/is a link is because of the old dashboard format. Given the update, it makes sense to me to have that action be just in the context menu. The title's never been styled as a link so it's not expected anyways. 

Does that make more sense?

**Comment by o-fernandez** [comment]
Ah, that makes sense. Thanks for clarifying and sorry I was confused. 

I like having an easy way to open the story in the editor without having to go to the context menu. It's the expected action, IMO.

Making the title "visibly clickable/not clickable" to account for the scenario where the user doesn't have edit capabilities for the story seems like a good thing to do, but I don't think it's a P0/critical item. We may want to make this a broader ticket that looks at "how do we communicate to users that they can't edit a story in the Dashboard". 

I'd say, we can track this in the backlog and resolve later? Let's get UX input as well.

**Comment by spacedmonkey** [comment]
While discussing this issue in sprint planning, I flagged that it isn't at all clear that title of the story is clickable. It has a hover, but that requires a user to hover on the link to see that. I keep finding myself in the dashboard and clicking on the poster image to edit a story. At the moment, this does nothing. Interfaces like youtube and netflix, have taught me to click on large thumbnails ( as this is touch / finger friendly ).

### YouTube
<img width="1521" alt="Screenshot 2021-08-13 at 12 15 02" src="https://user-images.githubusercontent.com/237508/129349345-cf062415-4045-4501-9710-bb6a8983a98e.png">

### Netflix
<img width="880" alt="Screenshot 2021-08-13 at 12 15 25" src="https://user-images.githubusercontent.com/237508/129349357-17eee900-3894-4cb3-87af-602b9a373b63.png">

We also have a carousel of stories in the gutenberg block, that looks similar and those posters are clickable. 

<img width="458" alt="Screenshot 2021-08-13 at 12 18 36" src="https://user-images.githubusercontent.com/237508/129349670-02b1a3ab-9d3f-4cc6-84fb-03343c549e24.png">

Instead of styling the title, just make the poster clickable. Then when the user clicks on the poster, if they do not have access, show then a dialog, saying they do not have permission. I would love UX to chim in on this one.

**Comment by BrittanyIRL** [comment]
I've asked Andrew to chime in since he's a designer and better equipped to handle this than either Jonny or myself.

**Comment by aaskedall** [comment]
I'm a fan of this idea for sure. Here is a quick mockup of how it could look. What do you think? 
![hover-idea](https://user-images.githubusercontent.com/4030811/129411471-10dd732b-9e53-46f9-a619-34fe958d5bee.gif)

**Comment by swissspidy** [comment]
> Then when the user clicks on the poster, if they do not have access, show then a dialog, saying they do not have permission. 

I don‘t think we need to go that far. Let‘s simply not make it clickable.

**Comment by BrittanyIRL** [comment]
@csossi more details are on the PR description, but QA for this is: 

### User Facing Changes 
If you have the ability to edit a story, you will get a purple border around grid items in the grid view of 'my stories' when you hover or focus on that item. If you do not have edit capabilities of a story then you will not see this nor will the story be clickable.

### Testing Instructions
Have a few stories in draft and published by at least an admin and contributor. Log in as both, hover and focus on stories. See that when you're a contributor you can only edit contributor stories and the published admin stories are not editable for you.

**Comment by csossi** [comment]
Verified in QA
